velox/common/compression/Compression.cpp

```diff
@@ -98,4 +99,151 @@ CompressionKind stringToCompressionKind(const std::string& kind) {
     VELOX_UNSUPPORTED("Not support compression kind {}", kind);
   }
 }
+
+void Codec::init() {}
+
+bool Codec::supportsGetUncompressedLength(CompressionKind kind) {
+  switch (kind) {
+    default:
+      return false;
+  }
+}
+
+bool Codec::supportsStreamingCompression(CompressionKind kind) {
+  switch (kind) {
+    case CompressionKind::CompressionKind_LZ4:
+      return true;
+    default:
+      return false;
+  }
+}
+
+bool Codec::supportsCompressFixedLength(CompressionKind kind) {
+  switch (kind) {
+    default:
+      return false;
+  }
+}
+
+folly::Expected<int32_t, Status> Codec::maximumCompressionLevel(
+    CompressionKind kind) {
+  return Codec::create(kind).then(
+      [](auto&& codec) { return codec->maximumCompressionLevel(); });
+}
+
+folly::Expected<int32_t, Status> Codec::minimumCompressionLevel(
+    CompressionKind kind) {
+  return Codec::create(kind).then(
+      [](auto&& codec) { return codec->minimumCompressionLevel(); });
+}
+
+folly::Expected<int32_t, Status> Codec::defaultCompressionLevel(
+    CompressionKind kind) {
+  return Codec::create(kind).then(
+      [](auto&& codec) { return codec->defaultCompressionLevel(); });
+}
+
+folly::Expected<std::unique_ptr<Codec>, Status> Codec::create(
+    CompressionKind kind,
+    const CodecOptions& codecOptions) {
+  if (!isAvailable(kind)) {
+    auto name = compressionKindToString(kind);
+    if (folly::StringPiece({name}).startsWith("unknown")) {
+      return folly::makeUnexpected(
+          Status::Invalid("Unrecognized codec '{}'", name));
+    }
+    return folly::makeUnexpected(
+        Status::Invalid("Support for codec '{}' not implemented.", name));
```

Comment:
Invalid status tells us it is unexpected, then why do we need to wrap it as folly unexpected?

Reply:
`folly::Expected` is like `arrow::Result`. It can represent either the returned value or error code.